src/rpm/common.go

```diff
@@ -0,0 +1,281 @@
+package rpm
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+
+	"github.com/tarantool/cartridge-cli/src/common"
+)
+
+type rpmValueType int32
+
+type rpmTagType struct {
+	ID    int
+	Type  rpmValueType
+	Value interface{}
+}
+
+type rpmTagSetType []rpmTagType
+
+type packedTagType struct {
+	Count int
+	Data  *bytes.Buffer
+}
+
+func (tagSet *rpmTagSetType) addTags(tags ...rpmTagType) {
+	*tagSet = append(*tagSet, tags...)
+}
+
+func packValues(values ...interface{}) *bytes.Buffer {
+	buf := bytes.NewBuffer(nil)
+
+	for _, v := range values {
+		binary.Write(buf, binary.BigEndian, v)
+	}
+
+	return buf
+}
+
+func packTag(tag rpmTagType) (*packedTagType, error) {
+	var packed packedTagType
+	packed.Data = bytes.NewBuffer(nil)
+
+	switch tag.Type {
+	case rpmTypeNull: // NULL
+		// XXX: It should be array of nil's ??
+	case rpmTypeChar: // CHAR
+		// XXX: It should be array of rune's or bytes ??
+	case rpmTypeBin: // BIN
+		byteArray, ok := tag.Value.([]byte)
+		if !ok {
+			return nil, fmt.Errorf("BIN value should be []byte")
+		}
+
+		packed.Count = len(byteArray)
+		for _, byteValue := range byteArray {
+			if _, err := io.Copy(packed.Data, packValues(byteValue)); err != nil {
+				return nil, err
+			}
+		}
+	case rpmTypeStringArray: // STRING_ARRAY
+		// value should be strings array
+		stringsArray, ok := tag.Value.([]string)
```

Comment:
Seems if you are in "rpmTypeStringArray" you have a guarantee that it could be casted to `[]string`. `!ok` check is redundant.

Reply:
BTW could it be rewritten as `tag.Value.(rpmTypeStringArray)`?

Reply:
No (discussed)